jwst/resample/resample_spec_step.py

```diff
@@ -36,28 +40,35 @@ def process(self, input):
         else:
             input_models = input_new
 
+        # Get the drizpars reference file
         for reftype in self.reference_file_types:
             ref_filename = self.get_reference_file(input_models[0], reftype)
 
         if ref_filename != 'N/A':
             self.log.info('Drizpars reference file: {}'.format(ref_filename))
             kwargs = self.get_drizpars(ref_filename, input_models)
         else:
-            # Deal with NIRSpec which currently has no default drizpars reffile
+            # Deal with NIRSpec, which currently has no default drizpars reffile
             self.log.info("No NIRSpec DIRZPARS reffile")
             kwargs = self._set_spec_defaults()
             kwargs['blendheaders'] = self.blendheaders
 
+        # Call resampling
         self.drizpars = kwargs
         if isinstance(input_models[0], MultiSlitModel):
             result = self._process_multislit(input_models)
 
         elif len(input_models[0].data.shape) != 2:
             # resample can only handle 2D images, not 3D cubes, etc
             raise RuntimeError("Input {} is not a 2D image.".format(input_models[0]))
+
         else:
             # result is a SlitModel
             result = self._process_slit(input_models)
+
+        # Update ASNTABLE in output
+        result.meta.asn.table_name = input_models[0].meta.asn.table_name
```

Comment:
Yes, because the table name got put into the individual input models in the recent change to `calwebb_spec3` (before they get sent to `resample_spec`). And I've verified that ASNTABLE is now showing up with the correct value in all `calwebb_spec3` products (source-based cal, s2d, x1d, c1d) by running all the various flavors of regtests that we have for spec2 and spec3.